pkg/providers/pricing/controller.go

```diff
@@ -50,9 +54,17 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) corecontrolle
 
 func (c *Controller) updatePricing(ctx context.Context) error {
 	work := []func(ctx context.Context) error{
-		c.pricingProvider.UpdateOnDemandPricing,
 		c.pricingProvider.UpdateSpotPricing,
 	}
+
+	// we try to update on demand pricing only if we are not in isolated VPC, as
+	// AWS Price list api might not be accessible for isolated VPCs
+	if !options.FromContext(ctx).IsolatedVPC {
+		work = append(work, c.pricingProvider.UpdateOnDemandPricing)
+	} else {
+		logging.FromContext(ctx).Infof("assuming isolated VPC, on-demand pricing information will not be updated")
```

Comment:
Is there a way to reduce spam to ensure that we only log this log once on startup or, minimally, that we only log it once per update to the pricing provider. My worry is that this is going to spam a lot if there is an error, where-as our pricing provider ensures that the log only comes every 12 hours using the change monitor. 

An alternative suggestion: What if we consider encapsulating the isolated VPC logic inside of the pricing provider and still call `c.pricingProvider.UpdateOnDemandPricing()`?

Reply:
Updated based on what I understood from your inputs. I am not sure if this is what you meant. With this change, log line should be added when reconciliation happens ( every 12 hours ). Let me know if this is not what you had in mind.